File: LC/python/Valid Parentheses.py
class Solution:
    # @return a boolean
    def isValid(self, s):
        stack = []
        last = ""
        
        for i in range(0,len(s)):
            if s[i] == "(" or s[i] == "[" or s[i] == "{":
                stack.append(s[i])
            else:
                if stack == []:
                    return False
                elif (s[i] == ")" and stack.pop() != "(") or (s[i] == "]" and stack.pop() != "[") or (s[i] == "}" and stack.pop() != "{"):
                    return False
                
        return not stack
                
        
        
        
        

File: LC/java/Valid Parentheses.java
public class Solution {
    public boolean isValid(String s) {
        Stack<Character> stack = new Stack<Character>();
        char tmp;
        for(int i = 0; i < s.length(); i++){
            if (s.charAt(i) == '(' || s.charAt(i) == '[' || s.charAt(i) == '{'){
                stack.push(s.charAt(i));
            }
            else{
                if(stack.isEmpty()){
                    return false;
                }
                tmp = stack.pop();
                if ((tmp == '(' && s.charAt(i) == ')') || (tmp == '[' && s.charAt(i) == ']') || (tmp == '{' && s.charAt(i) == '}')) ;
                else{
                    return false;
                }
            }
        }
        if(stack.isEmpty()){
            return true;
        }
        else{
            return false;
        }
    }
}